src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/TypeDefinitionRocks.cs

```diff
@@ -64,7 +64,11 @@ public static bool ImplementsInterface (this TypeDefinition type, string interfa
 		public static string GetPartialAssemblyQualifiedName (this TypeReference type)
 		{
 			TypeDefinition def = type.Resolve ();
-			return string.Format ("{0}, {1}", type.FullName, (def ?? type).Module.Assembly.Name.Name);
+			return string.Format ("{0}, {1}",
+					// Cecil likes to use '/' as the nested type separator, while
+					// Reflection uses '+' as the nested type separator. Use Reflection.
+					type.FullName.Replace ('/', '+'),
```

Comment:
Don't forget about the [previous question](https://github.com/xamarin/java.interop/pull/227#discussion_r157644899):

`TypeDefinitionRocks.GetPartialAssemblyQualifiedName()` is used in xamarin-android, in [`src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs`](https://github.com/xamarin/xamarin-android/blob/cf185ad3b7234b714ffd16c19c18a76cb07561d4/src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs#L165), so be careful about semantics.

The `<GenerateJavaStubs/>` code is writing into `obj/$(Configuration)/acw-map.txt`, which is a file which is *also* used by the IDEs.

Will using `+` instead of `/` confuse things? ¯\_(ツ)_/¯.

Reply:
@sgmunn? What does Visual Studio for Mac do with `acw-map.txt` around nested types? Will using `+` instead of `/` -- e.g. `System.Environment+SpecialFolder` instead of `System.Environment/SpecialFolder` -- break anything? Is that even knowable?

Reply:
It looks like this file would contain two different characters for nested classes right now:
https://github.com/xamarin/xamarin-android/blob/595186928ae4d79fab396dc05e7dc2750e808544/src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs#L165-L166

* `GetPartialAssemblyQualifiedName` returns `System.Environment/SpecialFolder` for the type portion
* `GetAssemblyQualifiedName` returns `System.Environment+SpecialFolder` for the type portion

Since `xamarin-android` will use `Type.GetType` at runtime, we need the `+` symbol in this PR: https://msdn.microsoft.com/en-us/library/w3f99sx1(v=vs.110).aspx#Remarks

But I bet the IDE might be using this in combination with something else that expects a `/`, like Mono.Cecil...

Maybe it is best to add an optional `nestedClassDelimiter` parameter that would keep the `acw-map.txt` file the same?